l10n_ch_invoice_reports/models/mail_template.py

```diff
@@ -0,0 +1,64 @@
+# Copyright 2021 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+import base64
+
+from odoo import _, models
+
+
+class MailTemplate(models.Model):
+    _inherit = "mail.template"
+
+    def generate_email(self, res_ids, fields=None):
+        """ Method overridden in order remove QR/ISR payslips and use
+            Invoice report with payslip generated by this module
+        """
+
+        result = super().generate_email(res_ids, fields)
+
+        multi_mode = True
+        if isinstance(res_ids, int):
+            res_ids = [res_ids]
+            multi_mode = False
+
+        res_ids_to_templates = self.get_email_template(res_ids)
+        for res_id in res_ids:
+            related_record = self.env[self.model_id.model].browse(res_id)
+            template = res_ids_to_templates[res_id]
+            if related_record._name == "account.move":
+                self.generate_invoice_attachment(
+                    result, template, res_id, related_record, multi_mode
+                )
+        return result
+
+    def generate_invoice_attachment(
+        self, result, template, res_id, related_record, multi_mode
+    ):
+        result[res_id]["attachments"] = False
+        self._render_template(template.report_name, template.model, res_id)
+        new_attachments = []
+        # We add an optional attachment from mail template (if set)
+        if self.report_template:
+            report_name = self.report_template.attachment
+            if (
+                self.report_template.report_name
+                == "l10n_ch_invoice_reports.account_move_payment_report"
+            ):
+                # force translation of attachment's name according to partner's lang
+                self = self.with_context(lang=related_record.partner_id.lang)
+                report_name = _(
+                    "invoice_%s_with_payslip.pdf"
+                ) % related_record.name.replace("/", "_")
+            report_xml_id = self.report_template.xml_id
+            report_pdf = self.env.ref(report_xml_id).render_qweb_pdf([res_id])[0]
+            report_pdf = base64.b64encode(report_pdf)
+            new_attachments.append((report_name, report_pdf))
+        attachments_list = (
+            multi_mode
+            and result[res_id].get("attachments", False)
+            or result.get("attachments", False)
+        )
+        if attachments_list:
+            attachments_list.extend(new_attachments)
+        else:
+            result[res_id]["attachments"] = new_attachments
```

Comment:
Why don't you check for multi mode here?
Also, what do you do w/ attachment_list from here?

Reply:
correct me if i am wrong:
we call this method in iteration, every time for single res_id from [res_ids] list -   afaiu multi mode is when we process multiple invoices - what's the point with checking for that here? i think we don't need this attachment list at all, we should just care about result[res_id][attachment] content. 
Majority of this code is copied from other project to not have to discover the wheel once again, could be that I missed some part of the problem here.

Reply:
ok for `multi_mode` but then:
1. why are you computing multi_mode above if you don't need it at all?
2. the very 1st thing you do in this new method is to wipe the values of `result[res_id]["attachment"]` -> how can you have values here?
3. to make things more clear `generate_invoice_attachment` should return the NEW result for each res_id instead of messing around w/ the old ones. Also, not all the params seems relevant.

> Majority of this code is copied from other project to not have to discover the wheel once again, could be that I missed some part of the problem here.

Can you share this w/ me in private please?